src/windowsservercore/ltsc2019/webassembly/Dockerfile

```diff
@@ -34,3 +56,32 @@ RUN msiexec /i %TEMP%\nodejs.msi /quiet /passive /qn /norestart
 RUN npm install jsvu -g
 RUN npm exec -c "jsvu --os=win64 --engines=v8,spidermonkey"
 RUN setx PATH "%PATH%;%USERPROFILE%\.jsvu"
+
+# Install cmake
+ENV CMAKE_VERSION 3.20.3
+RUN curl -SL --output %TEMP%\cmake-win.zip https://github.com/Kitware/CMake/releases/download/v%CMAKE_VERSION%/cmake-%CMAKE_VERSION%-windows-x86_64.zip \
+    && mkdir C:\cmake \
+    && tar -C C:\cmake -zxf %TEMP%\cmake-win.zip \
+    && setx PATH "%PATH%;C:\cmake\cmake-%CMAKE_VERSION%-windows-x86_64\bin"
+
+# Download the Build Tools bootstrapper.
+ADD https://aka.ms/vs/16/release/vs_buildtools.exe vs_buildtools.exe
+
+# Visual Studio with .NET and C++ tools
+RUN mkdir C:\BuildTools
+
+RUN vs_buildtools.exe --quiet --wait --norestart --nocache \
+    --installPath C:\BuildTools \
+    --add Microsoft.VisualStudio.Workload.ManagedDesktopBuildTools \
+    --add Microsoft.VisualStudio.Workload.VCTools \
+    --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 \
+    --remove Microsoft.VisualStudio.Component.Windows10SDK.10240 \
+    --remove Microsoft.VisualStudio.Component.Windows10SDK.10586 \
+    --remove Microsoft.VisualStudio.Component.Windows10SDK.14393 \
+    --remove Microsoft.VisualStudio.Component.Windows81SDK \
+    --add Microsoft.VisualStudio.Component.Windows10SDK.17763 \
+|| IF "%ERRORLEVEL%"=="3010" EXIT 0
```

Comment:
You must have gotten this from https://docs.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2019#create-and-build-the-dockerfile.  I actually was just made aware of that doc and filed an [issue](https://github.com/MicrosoftDocs/visualstudio-docs/issues/6748) on that because VS Build Tools is already installed from the base image.  

Instead of installing a new instance of BuildTools, the existing installation should be modified with the desired components.

```Dockerfile
RUN start /w vs_buildtools.exe --quiet --wait --norestart --nocache modify \
    --installPath "%ProgramFiles(x86)%\Microsoft Visual Studio\2019\BuildTools" \
    --add Microsoft.VisualStudio.Workload.ManagedDesktopBuildTools \
    --add Microsoft.VisualStudio.Workload.VCTools \
    --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 \
    --remove Microsoft.VisualStudio.Component.Windows10SDK.10240 \
    --remove Microsoft.VisualStudio.Component.Windows10SDK.10586 \
    --remove Microsoft.VisualStudio.Component.Windows10SDK.14393 \
    --remove Microsoft.VisualStudio.Component.Windows81SDK \
    --add Microsoft.VisualStudio.Component.Windows10SDK.17763 \
|| IF "%ERRORLEVEL%"=="3010" EXIT 0
```

Reply:
Oh, and this should also delete vs_buildtools.exe at the end.

Reply:
Nice, that would simplify things a bit. Let me try that.

Reply:
For unknown reason it doesn't modify the existing BuildTools for me. I think I will need to collect the logs and see if I can spot anything there.

I will push what I have in case you want to take a look.

Reply:
What indicates that it is not working for you (i.e. how are you testing the installation)?

Reply:
I check for few files, like `'C:\Program Files (x86)\Windows Kits\10\bin\*\x64\mt.exe'`, which should come from the Win10 SDK or `C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Auxiliary\Build\vcvarsall.bat` which comes from C++ VC tools.